bitrise.yml

```diff
@@ -3,51 +3,221 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
+    after_run:
+      - test-without-xcconfig
+      - test-xcconfig-from-env
+      - test-xcconfig-from-input
+      - test-xcconfig-from-input-url
+      - test-cache-xcconfig-from-env
+
+  test-without-xcconfig:
+    before_run:
+      - setup_clean_repo
+    after_run:
+      - teardown
+    steps:
+      - change-workdir:
+          title: Switch working dir to test / _tmp dir
+          inputs:
+            - path: $ORIG_BITRISE_SOURCE_DIR/_tmp
+            - is_create_path: true
+      - path::./:
+          run_if: |-
+            {{enveq "$XCODE11_12" "false"}}
+          inputs:
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+      - script:
+          run_if: |-
+            {{enveq "$XCODE11_12" "false"}}
+          title: Check outputs
+          inputs:
+            - content: |-
+                #!/usr/bin/env bash
+                set -ex
+                if [ -z "$BITRISE_CACHE_INCLUDE_PATHS" ] ; then echo "BITRISE_CACHE_INCLUDE_PATHS env is empty" ; exit 1 ; fi ;
+                [[ ! -f $ORIG_BITRISE_SOURCE_DIR/_tmp/Carthage/Cachefile ]] && echo "$ORIG_BITRISE_SOURCE_DIR/_tmp/Carthage/Cachefile does not exist" && exit 1
+                exit 0
+
+  test-xcconfig-from-env:
+    before_run:
+      - setup_clean_repo
+      - apply_xcode_workaround
+    after_run:
+      - assert-outputs
+      - teardown
     steps:
-    - go-list:
-    - golint:
-    - errcheck:
+      - change-workdir:
+          title: Switch working dir to test / _tmp dir
+          inputs:
+            - path: $ORIG_BITRISE_SOURCE_DIR/_tmp
+            - is_create_path: true
+      - script:
+          inputs:
+            - content: |-
+                #!/bin/bash
+                envman add --key XCODE_XCCONFIG_FILE --value "$TMP_XCODE_XCCONFIG_FILE"
+      - path::./:
+          inputs:
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+
+  test-cache-xcconfig-from-env:
+    before_run:
+      - setup_clean_repo
+      - apply_xcode_workaround
     after_run:
-    - test
+      - assert-outputs
+      - teardown
+    steps:
+      - change-workdir:
+          title: Switch working dir to test / _tmp dir
+          inputs:
+            - path: $ORIG_BITRISE_SOURCE_DIR/_tmp
+            - is_create_path: true
+      - script:
+          inputs:
+            - content: |-
+                #!/bin/bash
+                envman add --key XCODE_XCCONFIG_FILE --value "$TMP_XCODE_XCCONFIG_FILE"
+      - path::./:
+          inputs:
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+      - path::./:
+          inputs:
+            - carthage_options: --platform ios --project-directory $BITRISE_SOURCE_DIR
+
+  test-xcconfig-from-input:
+    before_run:
+      - setup_clean_repo
+      - apply_xcode_workaround
+    after_run:
+      - assert-outputs
+      - teardown
+    steps:
+      - change-workdir:
```

Comment:
Isn't this done in `setup_clean_repo`?

Reply:
The one in `setup_clean_repo` changes to `$ORIG_BITRISE_SOURCE_DIR` this changes to `$ORIG_BITRISE_SOURCE_DIR/_tmp`